### PR TITLE
Use `ark_ff::batch_inversion` for point normalization

### PR DESCRIPTION
**Pull request by Pratyush**
## Description

Reuses existing batch inversion code for batch normalization. Enables better parallelization, at the expense of some more allocation.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: ec/src/models/twisted_edwards_extended.rs [hunk 1]
```diff
@@ -444,51 +444,19 @@ impl<P: Parameters> ProjectiveCurve for GroupProjective<P> {
...
         // Perform affine transformations
-        v_iter.filter(|g| !g.is_normalized()).for_each(|g| {
-            g.x *= &g.z; // x/z
-            g.y *= &g.z;
-            g.t *= &g.z;
-            g.z = P::BaseField::one(); // z = 1
-        });
+        ark_std::cfg_iter_mut!(v)
+            .zip(z_s)
+            .filter(|(g, _)| !g.is_normalized())
```
Comment (ValarDragon) [inline ec/src/models/twisted_edwards_extended.rs:453]: Should the filter and zip be swapped?

Comment (ValarDragon) [reply ec/src/models/twisted_edwards_extended.rs:453]: Oh this code is different than the SW case, where this one doesn't filter what goes into the batch inversion.

Comment (ValarDragon) [reply ec/src/models/twisted_edwards_extended.rs:453]: Can we make the two cases the same? (Either filter out the already normalized elements in both, or apply the batch inversion to everything?)

Comment (Pratyush) [reply ec/src/models/twisted_edwards_extended.rs:453]: It wasn't compiling for some reason in that order

Comment (ValarDragon) [reply ec/src/models/twisted_edwards_extended.rs:453]: Oh no, this is just something that changed after the commit, ok this makes sense.

**Review by ValarDragon — APPROVED** [review]
LGTM, glad we'll get parallel batch inversion speedups here!

I wonder if we should eventually restructure this to use split the vector into chunks, and operate on each chunk in parallel. (Using a serial batch inversion on that chunk)
That would keep the data being mutated in the final loop thread-local

**Comment by Pratyush** [comment]
I think that's possible if we don't unify the logic with `batch_inversion` (essentially if we duplicate the logic in `batch_inversion`)

**Comment by ValarDragon** [comment]
Yeah I think we would have to duplicate that. We probably should move that standard chunk setup code to an ark-std macro

**Pull request merged**